editor/components/block-list/multi-controls.js

```diff
@@ -7,19 +7,29 @@ import { first, last } from 'lodash';
  * WordPress dependencies
  */
 import { withSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import BlockMover from '../block-mover';
 import BlockSettingsMenu from '../block-settings-menu';
+import NavigableToolbar from '../navigable-toolbar';
+import BlockSwitcher from '../block-switcher';
 
 function BlockListMultiControls( { multiSelectedBlockUids, rootUID, isSelecting, isFirst, isLast } ) {
 	if ( isSelecting ) {
 		return null;
 	}
 
 	return [
+		<NavigableToolbar
+			className="editor-block-contextual-toolbar"
+			aria-label={ __( 'Block Toolbar' ) }
+			key="toolbar"
+		>
+			<BlockSwitcher uids={ multiSelectedBlockUids } />
```

Comment:
I think a more appropriate place would be to move to the `BlockToolbar` component which will make it appear at the right position depending on the preference `Fix toolbar to top`

Reply:
Done! This required that the `BlockToolbar` became aware of multi-block selection, but it cut down on the repeated toolbars, and is a good point to insert multi-block toolbar controls :)